ravenframework/Optimizers/RavenSampled.py

```diff
@@ -301,8 +301,16 @@ def localFinalizeActualSampling(self, jobObject, model, myInput):
     # # testing suggests no big deal on smaller problem
     # the sign of the objective function is flipped in case we do maximization
     # so get the correct-signed value into the realization
-    if self._minMax == 'max':
-      rlz[self._objectiveVar] *= -1
+
+    if 'max' in self._minMax:
+      if not self._canHandleMultiObjective and len(self._objectiveVar) == 1:
+        rlz[self._objectiveVar[0]] *= -1
+      elif type(self._objectiveVar) == list:
+        for i in range(len(self._objectiveVar)):
+          if self._minMax[i] == 'max':
+            rlz[self._objectiveVar[i]] *= -1
+      else:
+        rlz[self._objectiveVar] *= -1
```

Comment:
self._objectiveVar is list only in GA. Other optimization code does not treat this variable as a list. This can be simplified later once everyone defines self._objectiveVar as a list.